### ISSUE TITLE
property setter is not respected when setting default value

### ISSUE THREAD

**Issue by nadako** [HaxeFoundation/haxe#7284]
```haxe
class Test {
  var field(default,set) = "hello";
  
  function new() {
    this.field = "hello again";
  }
  
  function set_field(value) return this.field = value;
  
  static function main() {
    trace(new Test().field);
  }
}
```

generated js code:
```js
var Test = function() {
	this.field = "hello";
	this.set_field("hello again");
};
```

first one should be `this.set_field("hello")`, right?

**Comment by ncannasse** [comment]
I'm not sure. You might as well want to define a default value without calling the setter that triggers some "value changed" code.

**Comment by Gama11** [comment]
Yeah, I think there might be quite a bit of code out there that relies on inits not triggering the setter.

**Comment by nadako** [comment]
i find this surprising, since to me, field default value is the same as setting it in the constructor, but there it calls the setter...

**Comment by Gama11** [comment]
I think the main issue is that without this behavior, there wouldn't be a way to initialize the underlying field without calling the setter?

**Comment by nadako** [comment]
i think for that we could use a meta like `@:bypassAccessor` (also see #5039)

**Comment by RealyUniqueName** [comment]
I always thought this is a designed behavior.
If you want to invoke setter, move the assignment of the default value to the constructor.

**Comment by nadako** [comment]
In my specific case I actually also don't need to trigger the setter, but still I expected it to call the setter. Maybe this is just something that should be documented better.

**Comment by Simn** [comment]
I always thought it was convenient that this bypasses the setter. And yes, this would likely break quite a big of code... So I agree the solution here is documentation.
